packages/block-editor/src/components/block-patterns-list/index.js

```diff
@@ -135,9 +135,12 @@ function BlockPatternList( {
 		>
 			{ blockPatterns.map( ( pattern ) => {
 				const isShown = shownPatterns.includes( pattern );
+				// User added unsynced patterns do not have a unique name so we use the id instead.
+				const key =
+					pattern.name === 'core/block' ? pattern.id : pattern.name;
```

Comment:
I think this component is `BlockPatternsList` is supposed to render patterns and patterns are expected to have unique names, so I think a better fix would be on the "caller" of this component, in the place where we transform "reusable blocks" to "patterns".

Reply:
I'm guessing that means `useUnsyncedPatterns`

Reply:
I included this change in [this alternative PR](https://github.com/WordPress/gutenberg/pull/51947)